pycalphad/mapping/zpf_equilibrium.py

```diff
@@ -0,0 +1,232 @@
+import copy
+import logging
+
+import numpy as np
+
+from pycalphad import calculate, variables as v
+from pycalphad.core.solver import Solver
+from pycalphad.core.composition_set import CompositionSet
+from pycalphad.property_framework.metaproperties import DormantPhase
+from pycalphad.core.constants import COMP_DIFFERENCE_TOL
+from pycalphad.property_framework.computed_property import JanssonDerivative
+
+from pycalphad.mapping.primitives import STATEVARS, Node, Point
+import pycalphad.mapping.utils as map_utils
+
+_log = logging.getLogger(__name__)
+
+def update_equilibrium_with_new_conditions(point: Point, new_conditions: dict[v.StateVariable, str], free_var: v.StateVariable = None):
+    """
+    Pretty much a copy of the old update with new conditions
+
+    Returns
+    -------
+    new_point - Point with updated composition sets and conditions
+        Checking whether the number of composition sets has changed will be checked later
```

Comment:
> Checking whether the number of composition sets has changed will be checked later

Maybe rephrase this to make it more clear that counting stable composition sets is out of scope for this function

Reply:
See above comment